ark/type/src/parser/tuple.ts

```diff
@@ -199,16 +212,32 @@ type inferTupleLiteral<
 	args,
 	result extends unknown[] = []
 > = def extends readonly [infer head, ...infer tail]
-	? inferDefinition<
-			head extends variadicExpression<infer operand> ? operand : head,
-			$,
-			args
-	  > extends infer element
-		? head extends variadicExpression
-			? element extends readonly unknown[]
-				? inferTupleLiteral<tail, $, args, [...result, ...element]>
-				: never
-			: inferTupleLiteral<tail, $, args, [...result, element]>
+	? parseEntry<
+			result["length"],
+			head extends variadicExpression<infer operand> ? operand : head
+	  > extends infer entryParseResult extends EntryParseResult
+		? inferDefinition<
+				entryParseResult["innerValue"] extends OptionalStringDefinition<
```

Comment:
The logic for dealing with the redundant ? case should be moved to parseEntry so we don't have to worry about it here.